components/brave_wallet/rust/lib.rs

```diff
@@ -162,9 +170,50 @@ pub struct Ed25519DalekSignature(Signature);
 struct Ed25519DalekExtendedSecretKeyResult(Result<Ed25519DalekExtendedSecretKey, Error>);
 struct Ed25519DalekSignatureResult(Result<Ed25519DalekSignature, Error>);
 struct Ed25519DalekVerificationResult(Result<(), Error>);
+struct Bech32DecodeResult(Result<Bech32DecodeValue, Error>);
+
 
 impl_result!(Ed25519DalekExtendedSecretKey, Ed25519DalekExtendedSecretKeyResult, ExtendedSecretKey);
 impl_result!(Ed25519DalekSignature, Ed25519DalekSignatureResult, Signature);
+
+impl From<bech32::Variant> for Bech32DecodeVariant {
+    fn from(v: bech32::Variant) -> Bech32DecodeVariant {
+        match v {
+            bech32::Variant::Bech32m => Bech32DecodeVariant::Bech32m,
+            bech32::Variant::Bech32 => Bech32DecodeVariant::Bech32,
+        }
+    }
+}
+
+impl Bech32DecodeResult {
```

Comment:
What is the reason of not using the macro `impl_result`?

Reply:
It doesn't compile 
`impl_result!(Bech32DecodeValue, Bech32DecodeResult, Bech32DecodeValue);`

`error[E0423]: expected function, tuple struct or tuple variant, found struct `Bech32DecodeValue`
   --> ../../brave/components/brave_wallet/rust/lib.rs:46:38
    |
21  | / macro_rules! impl_result {
22  | |     ($t:ident, $r:ident, $f:ident) => {
23  | |         impl $r {
24  | |             fn error_message(self: &$r) -> String {
...   |
46  | |                     Ok(v) => Self(Ok($t(v))),
    | |                                      ^^^^^
...   |
51  | |     };
52  | | }
    | |_- in this expansion of `impl_result!`
...
72  |       struct Bech32DecodeValue {
    |       ------------------------ `Bech32DecodeValue` defined here
...
173 |   struct Bech32DecodeResult(Result<Bech32DecodeValue, Error>);
    |   ------------------------------------------------------------ similarly named tuple struct `Bech32DecodeResult` defined here
...
178 |   impl_result!(Bech32DecodeValue, Bech32DecodeResult, Bech32DecodeValue);
    |   ---------------------------------------------------------------------- in this macro invocation
    |
help: use struct literal syntax instead
    |
46  |                     Ok(v) => Self(Ok(Bech32DecodeValue { hrp: val, data: val, variant: val })),
    |                                      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
help: a tuple struct with a similar name exists
    |
178 | impl_result!(Bech32DecodeResult, Bech32DecodeResult, Bech32DecodeValue);
`

Reply:
That is because you export `Bech32DecodeValue` directly and you shouldn't .
Please reference `Ed25519DalekExtendedSecretKey`